hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -828,19 +838,24 @@ update env ledger st ev = case (st, ev) of
     Effects [ClientEffect . ServerOutput.GetUTxOResponse headId $ getField @"utxo" $ getSnapshot confirmedSnapshot]
   (Open openState, NetworkEvent _ _ (ReqDec{transaction})) ->
     onOpenNetworkReqDec openState transaction
-  (Open OpenState{headId, coordinatedHeadState, currentSlot}, ClientEvent Decommit{decommitTx}) -> do
-    -- TODO: Spec: require U̅ ◦ decTx /= ⊥
-    requireValidDecommitTx $ \utxoToDecommit ->
-      Effects
-        [ ClientEffect ServerOutput.DecommitRequested{headId, utxoToDecommit}
-        , NetworkEffect ReqDec{transaction = decommitTx}
-        ]
+  (Open openState@OpenState{headId, coordinatedHeadState, currentSlot}, ClientEvent Decommit{decommitTx}) -> do
+    requireNoDecommitInFlight openState decommitTx $
+      -- TODO: Spec: require U̅ ◦ decTx /= ⊥
+      requireValidDecommitTx $ \utxoToDecommit ->
+        Effects
+          [ ClientEffect ServerOutput.DecommitRequestReceived{headId, utxoToDecommit}
+          , NetworkEffect ReqDec{transaction = decommitTx}
+          ]
    where
     -- TODO: Spec: require U̅ ◦ decTx /= ⊥
     requireValidDecommitTx cont =
       case applyTransactions ledger currentSlot confirmedUTxO [decommitTx] of
         Left (_, err) ->
-          Error $ RequireFailed $ DecommitTxInvalid{decommitTx, error = err}
+          Effects [ClientEffect ServerOutput.DecommitTxInvalid{headId, decommitTx}]
+            <> Error
+              ( RequireFailed $
+                  DecommitTxInvalid{decommitTx, error = err}
+              )
```

Comment:
Should: not need to have the `Error RequireFailed` outcome and the `DecommitTxInvalid` server output. The server output alone is enough (it gets logged and we don't do anything neither on `Effects` and `Error` outcome anyway)